src/System.Text.Json/tests/WritableJsonApiTests.JsonNumberUnitTests.cs

```diff
@@ -0,0 +1,473 @@
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(-17)]
+        [InlineData(3.14)]
+        [InlineData(-15.5)]
+        [InlineData(float.MinValue)]
+        [InlineData(float.MaxValue)]
+        public static void TestFloat(float value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetSingle(value);
+            Assert.Equal(value, jsonNumber.GetSingle());
+            Assert.True(jsonNumber.TryGetSingle(out float result));
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(-17)]
+        [InlineData(3.14)]
+        [InlineData(-15.5)]
+        [InlineData(double.MinValue)]
+        [InlineData(double.MaxValue)]
+        public static void TestDouble(double value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetDouble(value);
+            Assert.Equal(value, jsonNumber.GetDouble());
+            Assert.True(jsonNumber.TryGetDouble(out double result));
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(-17)]
+        [InlineData(sbyte.MinValue)]
+        [InlineData(sbyte.MaxValue)]
+        public static void TestSbyte(sbyte value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetSByte(value);
+            Assert.Equal(value, jsonNumber.GetSByte());
+            Assert.True(jsonNumber.TryGetSByte(out sbyte result));
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(ushort.MaxValue)]
+        public static void TestUInt16(ushort value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetUInt16(value);
+            Assert.Equal(value, jsonNumber.GetUInt16());
+            Assert.True(jsonNumber.TryGetUInt16(out ushort result));
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(uint.MaxValue)]
+        public static void TestUInt32(uint value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetUInt32(value);
+            Assert.Equal(value, jsonNumber.GetUInt32());
+            Assert.True(jsonNumber.TryGetUInt32(out uint result));
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(ulong.MaxValue)]
+        public static void TestUInt64(ulong value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetUInt64(value);
+            Assert.Equal(value, jsonNumber.GetUInt64());
+            Assert.True(jsonNumber.TryGetUInt64(out ulong result));
+            Assert.Equal(value, result);
+        }
+
+        [Fact]
+        public static void TestDecimal()
```

Comment:
Make this a theory like all the others, with different test values (max/min/0/etc.).

Reply:
I planned to, but `decimal.MaxValue` doesn't seem to be a constant expression. I get an error: "CS0182 An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type".

Reply:
Good point, it's a static readonly: https://docs.microsoft.com/en-us/dotnet/api/system.decimal.maxvalue?view=netframework-4.8

You can still make it a theory with MemberData instead of InlineData and use an IEunmerable<object[]> to define the test cases. For example:
https://github.com/dotnet/corefx/blob/93d7aa1c1737b6da29d04b78557215e18eb786d4/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L3201-L3203
https://github.com/dotnet/corefx/blob/93d7aa1c1737b6da29d04b78557215e18eb786d4/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L4448-L4459

As an aside, found some other interesting test cases:
https://github.com/dotnet/corefx/blob/93d7aa1c1737b6da29d04b78557215e18eb786d4/src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs#L689-L691